frontend/layouts/discover-page/component.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { FC, useEffect, useMemo, useState } from 'react';
+
+import cx from 'classnames';
+
+import { useRouter } from 'next/router';
+
+import DiscoverSearch from 'containers/discover-search';
+
+import LayoutContainer from 'components/layout-container';
+import { Paths } from 'enums';
+
+import { useProjectsList } from 'services/projects/projectService';
+
+import Header from './header';
+import Navigation from './navigation';
+import { DiscoverPageLayoutProps } from './types';
+
+export const DiscoverPageLayout: FC<DiscoverPageLayoutProps> = ({
+  screenHeightLg = false,
+  children,
+}: DiscoverPageLayoutProps) => {
+  const router = useRouter();
+  const { query } = router;
+
+  // We're keeping data in a state so that, while we're fetching new data,
+  // we can keep the existing one visible in the screen (and add a loading spinner).
+  const [projects, setProjects] = useState({ data: undefined, meta: undefined });
+
+  const queryParams = useMemo(
+    () => ({
+      page: parseInt(query.page as string) || 1,
+      search: (query.search as string) || '',
+    }),
+    [query]
+  );
+
+  const {
+    data: projectsData,
+    isFetching: isFetchingProjects,
+    isLoading: isLoadingProjects,
+  } = useProjectsList(queryParams);
+
+  useEffect(() => {
+    if (isFetchingProjects) return;
+    setProjects(projectsData);
+  }, [isFetchingProjects, projectsData]);
+
+  const stats = {
+    projects: projects?.meta?.total,
+    projectDevelopers: 0,
+    investors: 0,
+    openCalls: 0,
+  };
+
+  const { data, meta, loading } = useMemo(() => {
+    // TODO: Find a way to improve this.
+    if (router.pathname.startsWith(Paths.Projects))
+      return { ...projects, loading: isLoadingProjects };
+    // if (router.pathname.startsWith(Paths.ProjectDevelopers)) return projectDevelopers;
+    // if (router.pathname.startsWith(Paths.Investors)) return investors;
+    // if (router.pathname.startsWith(Paths.OpenCalls)) return openCalls;
+  }, [isLoadingProjects, projects, router.pathname]) || { data: [], meta: [] };
+
+  const handleSearch = (searchText: string) => {
+    router.push({ query: { ...queryParams, page: 1, search: searchText } });
```

Comment:
If the page gets server-rendered, it could be a good idea to use shallow routing: https://nextjs.org/docs/routing/shallow-routing